ui/raidboss/raidboss_options.ts

```diff
@@ -1,26 +1,30 @@
-import { BaseOptions } from '../../types/data';
+import { BaseOptions, RaidbossData } from '../../types/data';
 import UserConfig from '../../resources/user_config';
 import { Lang } from '../../resources/languages';
-import { TriggerAutoConfig } from '../../types/trigger';
+import { LooseTrigger, MatchesAny, TriggerAutoConfig, TriggerField, TriggerOutput } from '../../types/trigger';
 
 // This file defines the base options that raidboss expects to see.
 
 // Backwards compat for this old style of overriding triggers.
 // TODO: we should probably deprecate and remove this.
-type PerTriggerOption = Partial<{
+export type PerTriggerOption = Partial<{
   TextAlert: boolean;
   SoundAlert: boolean;
   SpeechAlert: boolean;
   GroupSpeechAlert: boolean; // TODO: we should remove this
   SoundOverride: string;
   VolumeOverride: number;
-  Condition: unknown; // TODO: trigger condition function
-  InfoText: unknown; // TODO: trigger function
-  AlertText: unknown; // TODO: trigger function
-  AlarmText: unknown; // TODO: trigger function
-  TTSText: unknown; // TODO: trigger function
+  Condition?: TriggerField<RaidbossData, boolean>; // TODO: trigger condition function
```

Comment:
If this is already Partial does it also need this question mark?

Reply:
It doesn't, apparently. I was having other issues with accessing this in popup-text while converting that adding the optional to the field resolved for some reason, but it seems fine removing it now on that branch, so I'll remove it with the next set of changes I guess?

Reply:
Also all these TODOs might be okay to remove now, unless they're referencing something else?

Reply:
There's also `// one giant TODO` below which is probably also okay to drop?